src/cdk/tree/tree.ts

```diff
@@ -358,6 +360,7 @@ export class CdkTreeNode<T> implements FocusableOption, OnDestroy {
 
   protected _setRoleFromData(): void {
```

Comment:
1. I added the isExpandable check to this function because it was an expedient way to avoid duplicating the logic. Do you recommend a rename to `setRoleAndExpandedFromData`, or another solution? Same goes for setRoleFromChildren.

2. This function is only called when the `data` setter in this class is invoked. In the CDK tree tests, this function is NOT invoked when you add a new node to the tree, so aria-expanded will not be updated. In the demo page (`yarn dev-app`) it does appear to be being called, so I think it might just be a bug in the tests. But I'm not sure, and could use some additional advice.

Reply:
For 1, I would rename this method to `_updateExpandedState` since, IIRC, you mentioned that the role should _always_ be treeitem anyway, right?

For 2, I think it's the same issue as https://github.com/angular/components/pull/17818